### PR TITLE
choropleth styling fixes

### PR DESCRIPTION
**Pull request by chejennifer**
Changes:
- handle window size changes
- show tooltip when hovering
- show the actual date instead of "latest" in the date dropdown
- center the legend

test-_12_

### CODE REVIEW

File: static/css/tools/choropleth.scss [hunk 1]
```diff
@@ -18,27 +18,15 @@
...
-}
-
-.clickable {
-  cursor: pointer;
-}
-
 #explore {
   float: left;
   width: 25%;
+  z-index: 5;
```
Comment (shifucun) [inline static/css/tools/choropleth.scss:29]: how does the z-index help here?

File: static/js/tools/choropleth/choropleth.tsx [hunk 2]
```diff
@@ -36,27 +43,40 @@ type StateType = {
...
+  constructor(props: PropsType) {
+    super(props);
+    this.svgContainerElement = React.createRef();
     const urlParams = new URLSearchParams(window.location.search);
     const isPerCapita =
       urlParams.has("pc") &&
       ["t", "true", "1"].includes(urlParams.get("pc").toLowerCase());
-    this.setState({
+    this.state = {
+      geoJson: [] as any,
```
Comment (shifucun) [inline static/js/tools/choropleth/choropleth.tsx:55]: why here needs "as any"?

File: static/js/tools/choropleth/choropleth.tsx [hunk 3]
```diff
@@ -164,28 +184,42 @@ class ChoroplethMap extends Component<PropsType, StateType> {
   private drawBlankGeoMap = (): void => {
     // Combine path elements from D3 content.
     const geojson = this.state["geoJson"];
-    const mapContent = d3
-      .select("#main-pane g.map")
-      .selectAll("path")
-      .data(geojson.features);
+    const container = document.getElementById("choropleth-body");
+    const heading = document.getElementById("heading");
+    const padding = 7;
```
Comment (shifucun) [inline static/js/tools/choropleth/choropleth.tsx:189]: this can move to the top of the file

File: static/css/tools/choropleth.scss (hunk 1 above)
Comment (chejennifer) [reply static/css/tools/choropleth.scss:29]: when the window gets too narrow, this prevents the choropleth map from showing up on top of the stat var menu

File: static/js/tools/choropleth/choropleth.tsx (hunk 2 above)
Comment (chejennifer) [reply static/js/tools/choropleth/choropleth.tsx:55]: I was just moving the original code, but I can change it

**Review by shifucun — APPROVED** [review]
Thanks for the fix!

**Pull request merged**